migrations/extract-data-record-content.js

```diff
@@ -0,0 +1,102 @@
+var _ = require('underscore'),
+    db = require('../db'),
+    async = require('async');
+
+var BATCH_SIZE = 100;
+var PERCENT_REPORT_CHUNKS = 10;
+
+// Copied from https://github.com/node-browser-compat/btoa/blob/master/index.js
+// because why import a library for one tiny function
+function btoa(str) {
+  var buffer;
+
+  if (str instanceof Buffer) {
+    buffer = str;
+  } else {
+    buffer = new Buffer(str.toString(), 'binary');
+  }
+
+  return buffer.toString('base64');
+}
+
+var attachmentIse = function(record) {
+  // XML forms which have not been migrated yet
+  if (record.content_type === 'xml' && record.content) {
+    record._attachments = record._attachments || {};
+    record._attachments.content = {
+      content_type: 'application/xml',
+      data: btoa(record.content)
+    };
+    delete record.content;
+
+    return record;
+  }
+};
+
+
+var incrementPercentTarget = function(percentTarget) {
+  return percentTarget + PERCENT_REPORT_CHUNKS;
+};
+
+module.exports = {
+  name: 'extract-data-record-content',
+  created: new Date(2017, 0, 6),
+  run: function(callback) {
+    console.log('Finding data_records...');
+    db.medic.view('medic', 'data_records', function(err, body) {
```

Comment:
Hmm good point. I had figured that just getting the list of ids was actually fine, but I suppose with larger DBs this might not be the same. I'll rejig it to have two layers of batching.

NB: this with percentages only works in this case because we're not filtering the view at all. In other situations we might need to batch load the view in chunks but still upfront to get percentages (somewhat ironically a % is only important when there is a lot of data, which in turn means loading ids upfront is a pita)